aws_lambda_powertools/utilities/parser/models/kafka.py

```diff
@@ -34,6 +34,14 @@ def decode_headers_list(cls, value):
                 header[key] = bytes(values)
         return value
 
+    # Validator to normalize the timestamp field
+    # Converts the provided timestamp value to a UTC datetime object
+    # See: https://github.com/pydantic/pydantic/issues/6518
+    @validator("timestamp", pre=True)
+    def coerce_timestamp(cls, value):
+        date_utc = datetime.fromtimestamp(int(value) / 1000, tz=timezone.utc)
+        return date_utc
```

Comment:
The Pydantic project will revert the behavior back to what we had in v1, so we can remove this validator after that.

Issue: https://github.com/pydantic/pydantic/issues/6518